cli/tests/environment-init.bats

```diff
@@ -160,32 +160,87 @@ function check_with_dir() {
   assert_equal "$init_system" "$NIX_SYSTEM"
 }
 
-@test "'flox init' sets up a working Python environment using requirements.txt" {
-  OWNER="owner"
-  NAME="name"
+# ---------------------------------------------------------------------------- #
 
-  echo "requests" > requirements.txt
+function checkPythonHook() {
   "$FLOX_BIN" init --auto-setup --name "$NAME"
+
   SHELL=bash "$FLOX_BIN" activate -- python -c "import requests"
   SHELL=zsh "$FLOX_BIN" activate -- python -c "import requests"
 
   floxhub_setup "$OWNER"
 
   "$FLOX_BIN" push --owner "$OWNER"
 
-  rm -rf .flox
+  "$FLOX_BIN" delete -f
 
   "$FLOX_BIN" pull "$OWNER/$NAME"
 
   SHELL=bash "$FLOX_BIN" activate -- python -c "import requests"
   SHELL=zsh "$FLOX_BIN" activate -- python -c "import requests"
 
-  rm -rf .flox
+  "$FLOX_BIN" delete -f
 
   SHELL=bash "$FLOX_BIN" activate --trust -r "$OWNER/$NAME" -- python -c "import requests"
   SHELL=zsh "$FLOX_BIN" activate --trust -r "$OWNER/$NAME" -- python -c "import requests"
 }
 
+# bats test_tags=init:python:requirements
+@test "'flox init' sets up a working Python environment using requirements.txt" {
+  export FLOX_FEATURES_INIT_PYTHON=true
+  OWNER="owner"
+  NAME="name"
+
+  echo "requests" >requirements.txt
+
+  checkPythonHook
+}
+
+# bats test_tags=init:python:pyproject:pip
+@test "'flox init' sets up a working Python environment using pyproject.toml and pip" {
+  export FLOX_FEATURES_INIT_PYTHON=true
+
+  OWNER="owner"
+  NAME="name"
+
+  cat <<-EOF >pyproject.toml
+  [project]
+  name = "name"
+  version = "0.0.0"
+  dependencies = [
+    "requests",
+  ]
+EOF
+
+  checkPythonHook
+}
+
+# bats test_tags=init:python:pyproject:poetry
+@test "'flox init' sets up a working Python environment using pyproject.toml and poetry" {
+  export FLOX_FEATURES_INIT_PYTHON=true
+
+  OWNER="owner"
+  NAME="name"
+
+  cat <<-EOF >pyproject.toml
+  [tool.poetry]
+  name = "name"
+  version = "0.0.0"
+  description = ""
+  authors = ["Your Name <you@example.com>"]
+
+  [tool.poetry.dependencies]
+  python = "^3.11"
+  requests = "^2.31.0"
+
+  [build-system]
+  requires = ["poetry-core"]
+  build-backend = "poetry.core.masonry.api"
+EOF
+
+  checkPythonHook
+}
```

Comment:
blocking: isn't this duplicate testing compared to what's in tests/python.bats? Seems like we can have tests in `cli/tests` that check that `init` behavior works overall (e.g., does `FLOX_ENV_CACHE` work with remote environments), but then checking each of pip+requirements/poetry/pip+pyproject.toml belongs in `tests/python.bats`. I could be talked into re-arranging stuff between `cli/tests` and `tests`, but it at least seems we don't need to check pip+requirements/poetry/pip+pyproject.toml in both places.

Reply:
Shall we go with just requirements here then?

Reply:
Yeah I think that's fine